### ISSUE TITLE
Make the samples directory easily runnable 

### ISSUE THREAD

**Issue by hossam-nasr** [Azure/azure-functions-durable-js#356]
See discussions here: https://github.com/Azure/azure-functions-durable-js/issues/334#issuecomment-1091919762. 

We should make the `samples` directory easily runnable. One of the challenges is that we have both TS and JS samples in the same directory. Perhaps we should split them into two directories, one for each language?

**Comment by davidmrdavid** [comment]
@hossam-nasr: Yeah it was a mistake to have both TS and JS samples in the same directory. Technically speaking, when the samples directory was created, it wasn't meant to be a runnable project but instead just a collection of code examples. Since then, its usage and expectations have evolved and now it makes sense to have them be runnable projects. 

I vote for splitting them up into JS and TS samples for runnability

**Comment by hossam-nasr** [comment]
@davidmrdavid Makes sense :) Also leaving a note here to remember to update the smoke test in the build pipeline to use all samples once this issue is resolved. Maybe we should even have two smoke tests, one for JS and one for TS.

**Comment by ejizba** [comment]
We should consider moving the samples out of this repo, into a place like https://github.com/Azure-Samples

And we should consider slimming down the samples - at the moment it's a bit intimidating how much code is in the "Samples" folder.

**Comment by davidmrdavid** [comment]
I'm not against moving the samples into `Azure-Samples`, although that would break the precedent established by other Durable Functions projects. For now, I'd rather hold off on that, if possible.

I +1 the idea of slimming down the samples. I think just splitting them up into JS and TS samples does quite a big of legwork there. Alternatively, just organizing them better would make it all easier to grok.

**Comment by hossam-nasr** [comment]
Closed in v3 of the package
